sample-app/src/hooks/useSynchronizedRequest.tsx

```diff
@@ -0,0 +1,24 @@
+import { useRef, useState } from "react";
+
+export function useSynchronizedRequest<T>(
+  executeRequest: (inputValue?: string) => Promise<T | undefined>
+): [
+    T | undefined,
+    (inputValue?: string) => Promise<T | undefined>
+  ]
+{
+  const networkIds = useRef({ latestRequest: 0, responseInState: 0 });
+  const [synchronizedResponse, setSynchronizedResponse] = useState<T>();
+  async function executeSynchronizedRequest (inputValue?: string): Promise<T | undefined> {
```

Comment:
if this is now handling generic requests, lets also make the param generic as well. like ..args? or an object so it can pass in whatever field to executeRequest to use

Reply:
updated to take in a generic param argument